ui/src/main/java/edu/wpi/grip/ui/python/PythonEditorController.java

```diff
@@ -0,0 +1,260 @@
+      } catch (PyException e) {
+        Alert malformed = new Alert(Alert.AlertType.ERROR);
+        malformed.setTitle("Error in script");
+        malformed.setHeaderText("There is an error in the python script");
+        malformed.getDialogPane().setContent(new Label(e.toString()));
+        malformed.showAndWait();
+        return false;
+      }
+      scriptFile = new File(PythonOperationUtils.DIRECTORY, fileName);
+    }
+    try {
+      Files.write(
+          scriptFile.getAbsoluteFile().toPath(),
+          codeArea.getText().getBytes(Charset.defaultCharset())
+      );
+      return true;
+    } catch (IOException e) {
+      logger.log(Level.WARNING, "Could not save to " + scriptFile, e);
+      Alert couldNotSave = new Alert(Alert.AlertType.ERROR);
+      couldNotSave.setTitle("Could not save custom operation");
+      couldNotSave.setContentText("Could not save to file: " + scriptFile);
+      couldNotSave.showAndWait();
+      return false;
+    }
+  }
+
+  @FXML
+  private void saveAs() {
+    FileChooser chooser = new FileChooser();
+    chooser.setInitialDirectory(PythonOperationUtils.DIRECTORY);
+    chooser.setTitle("Choose save file");
+    chooser.setSelectedExtensionFilter(
+        new FileChooser.ExtensionFilter("Custom GRIP operations", "*.py"));
+    String fileName = scriptFileName();
+    if (fileName != null) {
+      chooser.setInitialFileName(fileName);
+    }
+    File file = chooser.showSaveDialog(getWindow());
+    if (file == null) {
+      // Chooser was closed; no file selected
+      return;
+    }
+    scriptFile = file;
+    save();
+  }
+
+  @FXML
+  private void saveAndExit() {
+    if (save() && getScript() != null) {
+      // Don't exit if there's a problem with the script
+      exit();
+    }
+  }
+
+  @FXML
+  private void exit() {
+    getWindow().fireEvent(new WindowEvent(getWindow(), WindowEvent.WINDOW_CLOSE_REQUEST));
+  }
+
+  @FXML
+  private void openFile() {
+    FileChooser chooser = new FileChooser();
+    chooser.setInitialDirectory(PythonOperationUtils.DIRECTORY);
+    chooser.setTitle("Choose an operation to edit");
+    chooser.setSelectedExtensionFilter(
+        new FileChooser.ExtensionFilter("Custom GRIP operations", "*.py"));
+    File file = chooser.showOpenDialog(getWindow());
+    if (file == null) {
+      // Chooser was closed; no file selected
+      return;
+    }
+    try {
+      byte[] bytes = Files.readAllBytes(file.getAbsoluteFile().toPath());
+      String code = new String(bytes, Charset.defaultCharset());
+      codeArea.replaceText(code);
+    } catch (IOException e) {
+      logger.log(Level.WARNING, "Could not read file " + file, e);
+      Alert couldNotRead = new Alert(Alert.AlertType.ERROR);
+      couldNotRead.setTitle("Could not read custom operation");
+      couldNotRead.setContentText("Could not read from file: " + file);
+      couldNotRead.showAndWait();
+    }
+  }
+
+  @FXML
+  private void openWiki() {
```

Comment:
Take a look at how we did it for the [AboutDialog](https://github.com/WPIRoboticsProjects/GRIP/blob/946d8bc07ee424718ed9b28067e64fd183a7cca4/ui/src/main/java/edu/wpi/grip/ui/AboutDialogController.java#L19)